### ISSUE TITLE
[BUG] Parquet reader - Found no metadata for schema index

### ISSUE THREAD

**Issue by tgravescs** [NVIDIA/spark-rapids#6871]
**Describe the bug**
Customer reported issue when testing the 22.10 jar:

```
 Error operating ExecuteStatement: org.apache.spark.SparkException: Job aborted due to stage failure: Task 5098 in stage 470.0 failed 10 times, most recent failure: Lost task 5098.9 in stage 470.0 (TID 106902) (hadoop-dc6-13.nvdp.nvidia.com executor 468): ai.rapids.cudf.CudfException: cuDF failure at: /home/jenkins/agent/workspace/jenkins-spark-rapids-jni-release-3-cuda11/thirdparty/cudf/cpp/src/io/parquet/reader_impl.cu:430: Found no metadata for schema index
	at ai.rapids.cudf.Table.readParquet(Native Method)
	at ai.rapids.cudf.Table.readParquet(Table.java:1048)
	at com.nvidia.spark.rapids.MultiFileParquetPartitionReader.$anonfun$readBufferToTable$1(GpuParquetScan.scala:1673)
	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)
	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)
	at com.nvidia.spark.rapids.FilePartitionReaderBase.withResource(GpuMultiFileReader.scala:278)
	at com.nvidia.spark.rapids.MultiFileParquetPartitionReader.readBufferToTable(GpuParquetScan.scala:1672)
	at com.nvidia.spark.rapids.MultiFileCoalescingPartitionReaderBase.$anonfun$readToTable$1(GpuMultiFileReader.scala:900)
	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)
	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)
	at com.nvidia.spark.rapids.FilePartitionReaderBase.withResource(GpuMultiFileReader.scala:278)
	at com.nvidia.spark.rapids.MultiFileCoalescingPartitionReaderBase.readToTable(GpuMultiFileReader.scala:895)
	at com.nvidia.spark.rapids.MultiFileCoalescingPartitionReaderBase.$anonfun$readBatch$1(GpuMultiFileReader.scala:865)
	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)
	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)
	at com.nvidia.spark.rapids.FilePartitionReaderBase.withResource(GpuMultiFileReader.scala:278)
	at com.nvidia.spark.rapids.MultiFileCoalescingPartitionReaderBase.readBatch(GpuMultiFileReader.scala:847)
	at com.nvidia.spark.rapids.MultiFileCoalescingPartitionReaderBase.next(GpuMultiFileReader.scala:833)
	at com.nvidia.spark.rapids.PartitionIterator.hasNext(dataSourceUtil.scala:29)
	at com.nvidia.spark.rapids.MetricsBatchIterator.hasNext(dataSourceUtil.scala:46)
	at com.nvidia.spark.rapids.shims.GpuDataSourceRDD$$anon$1.$anonfun$hasNext$1(GpuDataSourceRDD.scala:53)
	at com.nvidia.spark.rapids.shims.GpuDataSourceRDD$$anon$1.$anonfun$hasNext$1$adapted(GpuDataSourceRDD.scala:53)
	at scala.Option.exists(Option.scala:376)
	at com.nvidia.spark.rapids.shims.GpuDataSourceRDD$$anon$1.hasNext(GpuDataSourceRDD.scala:53)
	at com.nvidia.spark.rapids.shims.GpuDataSourceRDD$$anon$1.advanceToNextIter(GpuDataSourceRDD.scala:78)
	at com.nvidia.spark.rapids.shims.GpuDataSourceRDD$$anon$1.hasNext(GpuDataSourceRDD.scala:53)
	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:37)
	at org.apache.spark.sql.rapids.GpuFileSourceScanExec$$anon$1.hasNext(GpuFileSourceScanExec.scala:442)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
	at com.nvidia.spark.rapids.CollectTimeIterator.$anonfun$hasNext$1(GpuExec.scala:193)
	at com.nvidia.spark.rapids.CollectTimeIterator.$anonfun$hasNext$1$adapted(GpuExec.scala:192)
	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)
	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)
	at com.nvidia.spark.RebaseHelper$.withResource(RebaseHelper.scala:26)
	at com.nvidia.spark.rapids.CollectTimeIterator.hasNext(GpuExec.scala:192)
	at com.nvidia.spark.rapids.AbstractGpuCoalesceIterator.hasNext(GpuCoalesceBatches.scala:288)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
	at com.nvidia.spark.rapids.CollectTimeIterator.$anonfun$hasNext$1(GpuExec.scala:193)
	at com.nvidia.spark.rapids.CollectTimeIterator.$anonfun$hasNext$1$adapted(GpuExec.scala:192)
	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)
	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)
	at com.nvidia.spark.RebaseHelper$.withResource(RebaseHelper.scala:26)
	at com.nvidia.spark.rapids.CollectTimeIterator.hasNext(GpuExec.scala:192)
	at scala.collection.Iterator$$anon$22.hasNext(Iterator.scala:1089)
	at com.nvidia.spark.rapids.CloseableBufferedIterator.hasNext(CloseableBufferedIterator.scala:38)
	at com.nvidia.spark.rapids.GpuBroadcastHashJoinExec.$anonfun$getBroadcastBuiltBatchAndStreamIter$2(GpuBroadcastHashJoinExec.scala:180)
	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)
	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)
	at com.nvidia.spark.rapids.GpuBroadcastHashJoinExec.withResource(GpuBroadcastHashJoinExec.scala:107)
	at com.nvidia.spark.rapids.GpuBroadcastHashJoinExec.$anonfun$getBroadcastBuiltBatchAndStreamIter$1(GpuBroadcastHashJoinExec.scala:179)
	at com.nvidia.spark.rapids.Arm.closeOnExcept(Arm.scala:87)
	at com.nvidia.spark.rapids.Arm.closeOnExcept$(Arm.scala:85)
	at com.nvidia.spark.rapids.GpuBroadcastHashJoinExec.closeOnExcept(GpuBroadcastHashJoinExec.scala:107)
	at com.nvidia.spark.rapids.GpuBroadcastHashJoinExec.getBroadcastBuiltBatchAndStreamIter(GpuBroadcastHashJoinExec.scala:178)
	at com.nvidia.spark.rapids.GpuBroadcastHashJoinExec.$anonfun$doExecuteColumnar$1(GpuBroadcastHashJoinExec.scala:215)
	at org.apache.spark.rdd.RDD.$anonfun$mapPartitions$2(RDD.scala:855)
	at org.apache.spark.rdd.RDD.$anonfun$mapPartitions$2$adapted(RDD.scala:855)
```

I currently don't have a repro case

**Comment by jlowe** [comment]
The fact that it works on PERFILE means the coalescing reader is combining two incompatible files into one.  I believe the problem is because the coalesing reader is checking the _names_ of columns to verify the schemas are compatible rather than checking the full read schemas (i.e.: not just names, but types and child types).

**Comment by jlowe** [comment]
@viadea confirmed that this is fixed by #6897.
